assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3066,6 +3080,21 @@ private Set<String> getSubmittersIdSet(List<AssignmentSubmission> submissions) {
         return submissions.stream().map(AssignmentSubmission::getSubmitters).flatMap(Set::stream).map(AssignmentSubmissionSubmitter::getSubmitter).collect(Collectors.toSet());
     }
 
+    public boolean isContentToolIdValid(Integer contentKey, Site site) {
+        if ( contentKey == null ) return true;
+        Long contentLong = contentKey.longValue();
+        Map<String, Object> content = ltiService.getContent(contentLong, site.getId());
+        if ( content == null ) return false;
+
+        Long toolKey = Long.valueOf(content.get(LTIService.LTI_TOOL_ID).toString());
+        if ( toolKey == null ) return false;
```

Comment:
`toolKey` can't be null here. If `content.get(LTIService.LTI_TOOL_ID)` would evaluate to `null` there would be a nullpointer on `toString()` or `NumberFormatException` if the string is empty.

Reply:
Yikes - that needs to be fixed!  Thanks